// yodlee-transactions/index.js
var path = require('path');
var fs = require("fs");
var globals = require('./globals');
var Promise = require('bluebird');
var https = require("https");
var request = Promise.promisify(require("request"));
Promise.promisifyAll(request);
module.exports = {
  yodlee : Yodlee
}

function Yodlee() {
  if(!(this instanceof Yodlee)) {
    return new Yodlee();
  }
}

//currently only working on USD
function getCategorySpending(yodleeTransactionObj) {
  var ts = yodleeTransactionObj.transaction;
  var result = {};
  ts.forEach(function(){
    var baseType = ts.baseType;
    var amount = ts.amount.amount;
    var category = ts.category;

    if (!result.hasOwnProperty(baseType)) {
      result.baseType = {};
    };

    if (!result[baseType].hasOwnProperty(category)) {
      result[baseType][category] =  0;
    }
    result[baseType][category] = result[baseType][category] + amount;
  });
  return result; //{Debit: {category1: amt, category2: amt } , Credit: {category1: amt, category2: amt } }
}

Yodlee.prototype.getTransactions = function(cobrandLogin, cobrandPassword, userName, userPassword) {
  globals.options.method = 'POST';
  globals.options.url = globals.baseUrl + globals.cobrandLoginURL;
  globals.options.headers = globals.headers;
  globals.options.form = {
    cobrandLogin : cobrandLogin,
    cobrandPassword : cobrandPassword
  }

  return request(globals.options, function(error, response, body) {
    var data = JSON.parse(body);
    globals.cobSessionToken = data.session.cobSession;
    globals.options.url = globals.baseUrl + globals.userLoginURL;
    globals.options.method = 'POST';
    globals.options.headers.Authorization = 'cobSession=' + globals.cobSessionToken
    //from arguments
    var yodleeUser = userName;
    var yodleePw = userPassword;
    globals.options.form =  {
      userLogin : yodleeUser,
      userPassword: yodleePw
    }
  //account, date to date
  //https://developer.api.yodlee.com:443/ysl/restserver/v1/transactions?accountId=account&fromDate=2014-05-05&toDate=2015-12-05&&&
  //general no dates
  //https://developer.api.yodlee.com:443/ysl/restserver/v1/transactions
    request(globals.options, function(error, response, body) {
      data = JSON.parse(body);
      globals.userSessionToken = data.session.userSession;
      console.log("cobSessionToken ", globals.cobSessionToken);
      console.log("userSessionToken", globals.userSessionToken);
      //Time to get the user transactions
      var transactionsOptions = {
          url: 'https://developer.api.yodlee.com:443/ysl/restserver/v1/transactions',
          method:  'GET',
          headers: {Authorization : "{cobSession=" + globals.cobSessionToken + "," + "userSession=" + globals.userSessionToken + "}"},
          form: ''
      }

      request(transactionsOptions)
        .then(function(data){
          return getCategorySpending(data);
        });
    });
  })
}
